docs/admin/cluster-components.md

```diff
@@ -0,0 +1,110 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+![WARNING](http://kubernetes.io/img/warning.png)
+![WARNING](http://kubernetes.io/img/warning.png)
+![WARNING](http://kubernetes.io/img/warning.png)
+
+<h1>PLEASE NOTE: This document applies to the HEAD of the source
+tree only. If you are using a released version of Kubernetes, you almost
+certainly want the docs that go with that version.</h1>
+
+<strong>Documentation for specific releases can be found at
+[releases.k8s.io](http://releases.k8s.io).</strong>
+
+![WARNING](http://kubernetes.io/img/warning.png)
+![WARNING](http://kubernetes.io/img/warning.png)
+![WARNING](http://kubernetes.io/img/warning.png)
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+# Kubernetes Cluster Admin Guide: Cluster Components
+
+This document outlines the various binary components that need to run to
+deliver a functioning Kubernetes cluster.
+
+## Master Components
+
+Master components are those that provide the cluster's control plane, make
+global decisions (i.e., scheduling) about the cluster, and detect and respond to
+cluster events (e.g., starting up a new pod when a replication controller's
+'replicas' field is unsatisfied).
+
+Master components could in theory be run on any node in the cluster. However,
+for simplicity, current set up scripts typically start all master components on
+the same VM. (But see [high-availability](availability.md) for an example
+multi-master-VM setup.)
+
+Even in the future, when Kubernetes is fully self-hosting, it will probably be
+wise to only allow master components to schedule on a subset of nodes, to limit
+co-running with user-run pods, reducing the possible scope of a
+node-compromising security exploit.
+
+### kube-apiserver
+
+```kube-apiserver``` is the global point of contact for the api. It is designed to
+scale horizontally (i.e., one scales it by running more of them).
+
+### etcd
+
+```etcd``` is used as Kubernetes' backing store. All cluster data is stored here.
+Proper administration of a Kubernetes cluster includes a backup plan for etcd's
+data.
+
+### kube-controller-manager
+
+```kube-controller-manager``` is a binary that runs controllers, or the background
+processes that handle routine tasks in the cluster.
+
+These controllers include:
+
+* Node Controller
+ * Responsible for noticing & responding when nodes go down.
+* Replication Controller
+ * Responsible for maintaining the correct number of pods for every replication
+   controller in the system.
```

Comment:
controller object (it's a bit circular to say the replication controller is responsible for maintaining the correct number of pods for every replication controller)